website/views.py

```diff
@@ -295,20 +295,9 @@ def resolve_guid(guid, suffix=None):
 
     # Retrieve guid data if present, error if missing
     try:
-        base_guid = guid.split('_v')[0]
-        base_guid_obj = Guid.objects.get(_id=base_guid.lower())
-
-        if base_guid_obj.is_versioned:
-            version = guid.split('_v')[1] if len(guid.split('_v')) > 1 else None
-            if version:
-                resource = base_guid_obj.versions.filter(version=version).first().referent
-            else:
-                # redirection to the same route, but with the specified version
-                version = base_guid_obj.versions.order_by('-version').first().version
-                new_guid = f'{base_guid}_v{version}'
-                return redirect(f'/{new_guid}/{suffix}' if suffix else f'/{new_guid}/', code=302)
-        else:
-            resource = Guid.objects.get(_id=base_guid.lower()).referent
+        resource = Guid.load_referent(guid)
+        if not guid == resource._id:
+            return redirect(f'/{resource._id}/{suffix}' if suffix else f'/{resource._id}/', code=302)
     except Guid.DoesNotExist:
         raise HTTPError(http_status.HTTP_404_NOT_FOUND)
```

Comment:
In addition `Guid.load_referent(guid)` doesn't throw exceptions as currently implemented. We should check if `resource` is None instead of `try ... except ...`

Reply:
In addition, let's refactor [newly merged code](https://github.com/CenterForOpenScience/osf.io/pull/10813/files#diff-e19218c1424dc830bfb6f2699703df289bfb2b27d3d3b7329c5580cf3605f8ceR209-R222) in `resolve_guid_download()`